powsybl-geo-data-server/src/main/java/com/powsybl/data/store/server/DataStoreServer.java

```diff
@@ -0,0 +1,207 @@
+public class DataStoreServer {
+
+    static final String API_VERSION = "v1";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DataStoreServer.class);
+
+    @Autowired
+    private NetworkGeoData networkGeoData;
+
+    @Autowired
+    private NetworkStoreService service;
+
+    @Autowired
+    private SubstationsRepository substationsRepository;
+
+    @Autowired
+    private LinesRepository linesRepository;
+
+    @Autowired
+    private LinesCustomRepository linesCustomRepository;
+
+    @GetMapping(value = "lines-graphics-with-pagination/{idNetwork}")
+    @ApiOperation(value = "Get Network Lines graphics", produces = "application/json")
+    @ApiResponses(value = {@ApiResponse(code = 200, message = "The list of lines graphics")})
+    public ResponseEntity<List<LineGraphic>> getLinesGraphicsWithPagination(@PathVariable UUID idNetwork, @RequestParam(name = "page") int page, @RequestParam(name = "size") int size) {
+        Network network = service.getNetwork(idNetwork);
+        return  ResponseEntity.ok().body(getLinesGraphicsElements(network, page, size));
+    }
+
+    @GetMapping(value = "lines-graphics/{idNetwork}")
+    @ApiOperation(value = "Get Network lines graphics", produces = "application/json")
+    @ApiResponses(value = {@ApiResponse(code = 200, message = "The list of lines graphics")})
+    public ResponseEntity<List<LineGraphic>> getLinesGraphics(@PathVariable UUID idNetwork) {
+        Network network = service.getNetwork(idNetwork);
+        return ResponseEntity.ok().body(new ArrayList<>(networkGeoData.getNetworkLinesCoordinates(network).values()));
+    }
+
+    @GetMapping(value = "substations-graphics/{idNetwork}")
+    @ApiOperation(value = "Get Network substations graphics", produces = "application/json")
+    @ApiResponses(value = {@ApiResponse(code = 200, message = "The list of substations graphics")})
+    public ResponseEntity<List<SubstationGraphic>> getSubstationsGraphic(@PathVariable UUID idNetwork) {
+        Network network = service.getNetwork(idNetwork);
+        return  ResponseEntity.ok().body(new ArrayList<>(networkGeoData.getSubstationsCoordinates(network).values()));
+    }
+
+    @GetMapping(value = "substations-graphics-with-pagination/{idNetwork}")
+    @ApiOperation(value = "Get Network substations graphics", produces = "application/json")
+    @ApiResponses(value = {@ApiResponse(code = 200, message = "The list of substations graphics")})
+    public ResponseEntity<List<SubstationGraphic>> getSubstationsGraphicsWithPagination(@PathVariable UUID idNetwork, @RequestParam(name = "page") int page, @RequestParam(name = "size") int size) {
+        Network network = service.getNetwork(idNetwork);
+        List<SubstationGraphic> substationGraphics = getSubstationsGraphicsElements(network, page, size);
+        return ResponseEntity.ok().body(substationGraphics);
+    }
+
+    private List<LineGraphic> getLinesGraphicsElements(Network network, int page, int size) {
```

Comment:
Would it be possible to move all related pagination code to a utility class and share it between lines and  substations?